app/Main.hs

```diff
@@ -1,32 +1,41 @@
+{-# LANGUAGE QuasiQuotes #-}
+{-# LANGUAGE TemplateHaskell #-}
+
 import System.Environment (getArgs)
 import System.Exit (exitFailure)
 
+import System.IO (hPutStrLn, stderr)
+
 import Data.Semigroup ((<>))
 import Data.Text (unpack)
 import qualified Options.Applicative as Opt
-import Options.Applicative ((<**>))
+import Options.Applicative ((<**>), many)
+
+import Control.Exception.Safe
+import PyF
 
 import Krank
 import Krank.Formatter
 
 data KrankOpts = KrankOpts {
-  codeFilePath :: FilePath
+  codeFilePaths :: [FilePath]
 }
 
-fileToParse :: Opt.Parser FilePath
-fileToParse = Opt.argument Opt.str (Opt.metavar "FILE")
+filesToParse :: Opt.Parser [FilePath]
+filesToParse = many (Opt.argument Opt.str (Opt.metavar "FILES..."))
 
 sample :: Opt.Parser KrankOpts
-sample = KrankOpts <$> fileToParse
+sample = KrankOpts <$> filesToParse
 
 opts :: Opt.ParserInfo KrankOpts
 opts = Opt.info (sample <**> Opt.helper)
   ( Opt.fullDesc
-  <> Opt.progDesc "Checks the comments in FILE"
+  <> Opt.progDesc "Checks the comments in FILES"
   <> Opt.header "krank - a comment linter / analytics tool" )
 
 main :: IO ()
 main = do
   options <- Opt.execParser opts
-  violations <- processFile . codeFilePath $ options
-  putStrLn . unpack . showViolations $ violations
+  (flip mapM_) (codeFilePaths options) $ \path -> do
```

Comment:
`flip` will go in my memory. did not know this one! 👍

Reply:
Technically you can also use `forM`, `forM_` from `Control.Monad` and `for` from `Data.Traversable` and `for_` from `Data.Foldable`. But I'm lazy and I use `flip mapM_`.